### PR TITLE
Pagination and column widths handling

### LINKED ISSUE #924: Enable page_by pagination for docx

### PR DESCRIPTION
**Pull request by Melkiades**
Fixes #924 and hopefully #925

It seems that the pagination needs a lot of additional work to function correctly. It may be better to work on it in the next PI. @shajoezhu fyi here pagination works but column widths and lpp/cpp estimations are a bit difficult to concretize consistently

### CODE REVIEW

**Comment by shajoezhu** [comment]
i start thinking again that we need another downstream package for exporting

**Comment by Melkiades** [comment]
> i start thinking again that we need another downstream package for exporting

The export functions are in {rtables} when available only for tables and {formatters} when for both tables and listings. Making a separate package would make sense if we wanted to cover more objects, but I do not see this happening soon. Also, exporters are relatively small; the larger function here is the flextable transformer, for which there are a lot of aesthetics that are hard to test. 

In summary, we already have a complex dependency tree, and we might want to keep dev exporters in {rtables} and full exporters (w/ listings capabilities) in {formatters}. Beside that I do not see too much the advantage of having exporters in a separate package atm. Also the risk of circular deps would be very real here, forcing us to move function around beyond what we need to. 

I am open to discussing this anyway! Maybe a pros and cons next PI? :)

**Comment by Melkiades** [comment]
> Fixes #924 and hopefully #925

I managed to make these work, but the estimation of the actual page dimension is very faulty. It will need more design for that

**Review by edelarua — APPROVED** [review]
This looks good to go!! It can be revisited at a later date if refinement is needed :)
